store/postgres/src/relational_queries.rs

```diff
@@ -867,24 +869,126 @@ impl<'a> fmt::Display for QueryFilter<'a> {
+                // the root table is always the parent table (because we support only one level of nesting)
+                let parent_column = self.table.column_for_field(field_name)?;
+                let parent_prefix = "c".to_string();
+                let child_table = self.layout.table_for_entity(entity)?;
+                let child_column = child_table.primary_key();
+                let child_prefix = format!("c_{}", field_name);
+
+                fn push_column_with_prefix(
+                    mut out: AstPass<Pg>,
+                    column: &str,
+                    prefix: &String,
+                ) -> QueryResult<()> {
+                    out.push_sql(format!("{}.", prefix).as_str());
+                    out.push_identifier(column)?;
+
+                    Ok(())
                 }
+
+                out.push_sql("\n");
+
+                out.push_sql("inner join ");
+                out.push_sql(child_table.qualified_name.as_str());
+                out.push_sql(format!(" {}", child_prefix).as_str());
+
+                out.push_sql(" on (");
+
+                // Join by ID
+                push_column_with_prefix(out.reborrow(), child_column.name.as_str(), &child_prefix)?;
+                out.push_sql(" = ");
+                push_column_with_prefix(
+                    out.reborrow(),
+                    parent_column.name.as_str(),
+                    &parent_prefix,
+                )?;
+
+                out.push_sql(" AND ");
+
+                // Match by block
+                push_column_with_prefix(out.reborrow(), BLOCK_RANGE_COLUMN, &child_prefix)?;
+                out.push_sql(" @> ");
+                out.push_bind_param::<Integer, _>(&block)?;
+
+                out.push_sql(" )");
+
+                self.visit_child_filters(block, child_filter, out)?;
             }
+            _ => {}
+        })
+    }
 
+    /// Creates a SQL fragment representing the table joins.
+    ///
+    ///   inner join child1 c1 ON ( c1.id = parent.child_id AND c1.block_range @> $block )
+    ///   inner join child2 c2 ON ( c2.id = parent.child_id AND c2.block_range @> $block )
+    ///
+    pub fn walk_joins_ast(&self, block: BlockNumber, mut out: AstPass<Pg>) -> QueryResult<()> {
+        out.unsafe_to_cache_prepared();
+        self.visit_child_filters(block, self.filter, out)
+            .map_err(|e| DieselError::QueryBuilderError(Box::new(e)))?;
+        Ok(())
+    }
+
+    fn valid_attributes(
+        filter: &'a EntityFilter,
+        table: &'a Table,
+        layout: &'a Layout,
+        child_filter_ancestor: bool,
+    ) -> Result<(), StoreError> {
+        use EntityFilter::*;
+        match filter {
+            And(filters) | Or(filters) => {
+                for filter in filters {
+                    Self::valid_attributes(filter, table, layout, child_filter_ancestor)?;
+                }
+            }
+            Child(_, entity, child_filter) => {
```

Comment:
This check should also make sure that the attribute name is valid for `table`, i.e., call `table.column_for_field(attr)?`